tests/test_intervals.py

```diff
@@ -349,3 +349,615 @@ def test_points_per_interval_slicer(test_data, ppi_params_and_ref):
+
+    """ 
+        
+    # n=2 include_max=True, value_range=(0, 5)
+    ref_intervals = [[1.2, 1.5, 2.4], [2.5, 2.6, 3.1, 3.5, 3.6, 4.0, 5.0]]
+    ref_references = [1.25, 3.75]
+    ref_boundaries = [[0, 2.5], [2.5, 5.0]]
+    number_slicer = NumberOfIntervalsSlicer(2, value_range=(0, 5), min_n_points=1)
+    my_slices, my_references, my_boundaries = number_slicer.slice_(test_data)
+    my_intervals = [test_data[slice_] for slice_ in my_slices]
+    np.testing.assert_array_equal(my_references, ref_references)
+    assert len(my_intervals) == len(ref_intervals)
+    for i in range(len(my_intervals)):
+        np.testing.assert_array_equal(my_intervals[i], ref_intervals[i])
+    for i in range(len(my_boundaries)):
+        np.testing.assert_array_equal(my_boundaries[i], ref_boundaries[i])
+
+
+def test_number_of_interval_slicer_numberpoints2_include_max_reference_median(test_data):
+    """
+    Function to compare characterisics of reference interval generated by
+    NumberOfIntervalSlicer and manually generated interval.
+
+    Parameters
+    ----------
+    test_data : numpy array
+        Data to test the interval slicer.
+ 
+    Returns
+    -------
+    None.
+
+    """ 
+    
+    # n=2 include_max=True, reference=np.median
+    ref_intervals = [[1.2, 1.5, 2.4, 2.5, 2.6], [3.1, 3.5, 3.6, 4.0, 5.0]]
+    ref_references = [2.4, 3.6]
+    ref_boundaries = [[1.2, 3.1], [3.1, 5.0]]
+    number_slicer = NumberOfIntervalsSlicer(2, reference=np.median, min_n_points=1)
+    my_slices, my_references, my_boundaries = number_slicer.slice_(test_data)
+    my_intervals = [test_data[slice_] for slice_ in my_slices]
+    np.testing.assert_array_equal(my_references, ref_references)
+    assert len(my_intervals) == len(ref_intervals)
+    for i in range(len(my_intervals)):
+        np.testing.assert_array_equal(my_intervals[i], ref_intervals[i])
+    for i in range(len(my_boundaries)):
+        np.testing.assert_almost_equal(my_boundaries[i], ref_boundaries[i])
+    
+
+def test_number_of_interval_slicer_numberpoints3_numberintervals3(test_data):
+    """
+    Function to compare characterisics of reference interval generated by
+    NumberOfIntervalSlicer and manually generated interval.
+
+    Parameters
+    ----------
+    test_data : numpy array
+        Data to test the interval slicer.
+ 
+    Returns
+    -------
+    None.
+
+    """ 
+    
+    # n=3 include_max=True, min_n_points=3, min_n_intervals=3
+    ref_intervals = [[1.2, 1.5, 2.4], [2.5, 2.6, 3.1, 3.5, 3.6]]
+    ref_width = (5 - 1.2) / 3
+    ref_references = [1.2 + ref_width / 2, 1.2 + 3 * ref_width / 2]
+    ref_boundaries = [(c - ref_width / 2, c + ref_width / 2) for c in ref_references]
+    number_slicer = NumberOfIntervalsSlicer(3, min_n_points=3, min_n_intervals=3)
+    try:
+        my_slices, my_references, my_boundaries = number_slicer.slice_(test_data)
+    except RuntimeError:
+        pass  # we expect a RuntimeError if there are less than min_n_intervals intervals
```

Comment:
Please start the comment with a capital letter and end it with a dot (full sentence).